engine-tests/src/test_utils/random.rs

```diff
@@ -0,0 +1,81 @@
+use crate::prelude::U256;
+use crate::test_utils::{self, solidity, AuroraRunner, Signer};
+use aurora_engine::transaction::legacy::TransactionLegacy;
+use ethabi::Constructor;
+
+const DEFAULT_GAS: u64 = 1_000_000_000;
+
+pub(crate) struct RandomConstructor(pub solidity::ContractConstructor);
+
+impl RandomConstructor {
+    pub fn load() -> Self {
+        Self(solidity::ContractConstructor::compile_from_extended_json(
+            "../etc/eth-contracts/artifacts/contracts/test/Random.sol/Random.json",
+        ))
+    }
+
+    pub fn deploy(&self, nonce: u64) -> TransactionLegacy {
+        let data = self
+            .0
+            .abi
+            .constructor()
+            .unwrap_or(&Constructor { inputs: vec![] })
+            .encode_input(self.0.code.clone(), &[])
+            .unwrap();
+
+        TransactionLegacy {
+            nonce: nonce.into(),
+            gas_price: Default::default(),
+            gas_limit: U256::from(DEFAULT_GAS),
+            to: None,
+            value: Default::default(),
+            data,
+        }
+    }
+}
+
+impl From<RandomConstructor> for solidity::ContractConstructor {
+    fn from(c: RandomConstructor) -> Self {
+        c.0
+    }
+}
+
+pub(crate) struct Random {
+    contract: solidity::DeployedContract,
+}
+
+impl Random {
+    pub fn random_seed(&self, runner: &mut AuroraRunner, signer: &mut Signer) -> Option<U256> {
```

Comment:
Seems unnecessary to return an `Option` here when we unwrap all the errors except the return length. Since this is only for tests I think it is fine to return simply `U256` (panicking if the result length is not 32 bytes).

As a side note, I don't think `U256` is the right type. Yes, a 32-byte array is returned but semantically I don't think we would normally interpret it as a number. `H256` or even just `[u8; 32]` would be better in my opinion.

Reply:
Refactored everything to use `H256`.